trieste/acquisition/function.py

```diff
@@ -706,6 +706,9 @@ def __init__(
         self._objective_tag = objective_tag
         self._constraint_builder = constraint_builder
         self._min_feasibility_probability = min_feasibility_probability
+        self._constraint_fn: Optional[AcquisitionFunction] = None
+        self._improvement_fn: Optional[AcquisitionFunction] = None
+        self._constrained_fn: Optional[AcquisitionFunction] = None
```

Comment:
Can these have a bit more distinctive names? It is quite hard to figure out which of two `_constrain...` is being used while reading the code.

Reply:
ok, noted, will hold off further reviewing for a bit